test: add minimal test infra

antongolub opened the pull request.

antongolub committed:
test: add minimal test infra

wobsoriano commented:
Curious as to why `uvu` is used instead of built in `bun:test`?

```js
import { describe, expect, it, test } from "bun:test"
```

antongolub committed:
test: add test-utils

antongolub commented:
> Curious as to why uvu is used instead of built in bun:test?  

There're sveral reasons for now (just a personal vision):
* runner: `uvu '/any/.+/valid-regex-pattern/'`
* `test.skip` & `test.only` and other helpers that are useful in TDD.
* summary across the all test suites

Thanks for highlighting. Added a couple of helpers to make it easier to migrate to `bun:test` in future.

mattreid1 commented:
Ideally I'd like to use to the native Bun test runner, but given it's still a work in progress, I really like this approach with the helpers! I'll probably merge this tomorrow when I have a little more time to look over it. Thanks!

mattreid1 committed:
Delete ci.yaml

I don't want to add CI *just* yet. Definitely something to add in the future though!
